### DISCUSSION TITLE
Not able to declare best solution even after actually finding it

### DISCUSSION THREAD

**Discussion by cyasing** [anyoptimization/pymoo#420] [Q&A] [answered]
I am trying for a simple GA script, where in I do see that the X values converge to the ideal, even the F values reach the minimum (verbose = True and printing X values via evaluate function from my problem class), and yet the program just keeps on running without actually printing res.X and res.F. What could be potential reasons for it?

**Comment by cyasing** [comment]
> For reference, here's the output:
![image](https://user-images.githubusercontent.com/59028917/229523177-9ec4cd64-19b7-4ce8-a12d-125b6622e940.png)

**Reply by cyasing** [reply]
The first array represents X, while the single line outputs are all outputs from pymoo. It's the same for each iteration. The X converge to 1, and yet at the 12th iteration, it did not print anything for a long time (15 minutes when each iteration took a minute.)

**Comment by blankjul** [comment]
If it is a simple example please provide the source code so that we know what you are talking about.

Are you using the Callback to print the values? https://pymoo.org/interface/callback.html
You can access the current optimum by `algorithm.opt` which is a `Population` object.

**Reply by cyasing** [reply]
It isn't a simple example, per se. I am using a physics based model (specifically PyBaMM's battery model) that takes in one input, and in the evaluate function solves the model for all 10 of these inputs and then the objective values are obtained. Would you be interested in viewing the program?

**Reply by cyasing** [reply]
I am not using a callback, just simple print statement in evaluate function.

**Reply by cyasing** [reply]
Can you tell me where to use this `algorithm.opt` other than callbacks? Or a simpler callback to just have F and X values printed properly?

**Reply by blankjul** [reply, marked as answer]
just use `algorithm.opt.get("X")` and `algorithm.opt.get("F")`. You can use the methods an ANY algorithm object (not only in callback)

**Reply by cyasing** [reply]
Thanks!
